# Auditoria de divergências

Triagem de 24 features. Revisão individual completa de autenticação; demais linhas mantêm grau de certeza e pendências. Veja RELATORIO.md. Nenhuma sugestão altera a fonte.

In [1]:
from pathlib import Path
import json, pandas as pd
from IPython.display import display
PASTA = Path.cwd()
if not (PASTA/'resumo_auditoria.json').exists():
    PASTA = PASTA/'auditoria_divergencias_20260917'
resumo=json.loads((PASTA/'resumo_auditoria.json').read_text())
display(pd.Series({k:v for k,v in resumo.items() if not isinstance(v,dict)}))

escopo               Triagem de todas as divergências salvas; leitu...
antes                                                             4260
depois                                                            3757
concordancia                                                 88.378495
casos_regressao                                                    187
testes_sinteticos                                                  144
dtype: object

## Antes e depois
Concordância com anotações não equivale a acurácia; o original não é padrão-ouro.

In [2]:
display(pd.read_csv(PASTA/'resumo_por_feature.csv'))

,feature,divergencias_antes,divergencias_depois
0,solicita_senha,13,13
1,solicita_codigo_autenticacao,121,110
2,solicita_dados_bancarios,1,0
3,solicita_dados_pessoais,70,51
4,solicita_dados_cartao,19,19
5,solicita_atualizacao_cadastro,66,65
6,solicita_pagamento,201,170
7,solicita_clique,472,408
8,solicita_ligacao,47,13
9,menciona_bloqueio_conta,118,108


## Código de autenticação
110 prováveis erros originais; 11 falhas do script corrigidas. CSV completo disponível para contestar cada parecer.

In [3]:
auth=pd.read_csv(PASTA/'analise_codigo_autenticacao.csv')
display(auth.situacao_analise.value_counts())
display(auth[['registro_original','Mensagem','valor_original','valor_gerado_revisado','justificativa_analise']].head(12))

situacao_analise
provavel_erro_original    110
falha_script_corrigida     11
Name: count, dtype: int64

,registro_original,Mensagem,valor_original,valor_gerado_revisado,justificativa_analise
0,20,C?digo de login em Ativas Farma: 0564 @ativasf...,1,0,"Entrega ou informa um código, sem solicitar se..."
1,81,Use 326829 para validar seu telefone na OLX. P...,1,1,Solicita uso/informação/digitação do código em...
2,82,3m1Ew/Zt9XJAmazon: Your code is 499696. Don t ...,1,0,"Entrega ou informa um código, sem solicitar se..."
3,83,Amazon: Your code is 816874. Don t share it. I...,1,0,"Entrega ou informa um código, sem solicitar se..."
4,84,G-106571 é seu código de verificação do Google.,1,0,"Entrega ou informa um código, sem solicitar se..."
5,85,G-864638 é seu código de verificação do Google.,1,0,"Entrega ou informa um código, sem solicitar se..."
6,86,G-733039 é seu código de verificação do Google.,1,0,"Entrega ou informa um código, sem solicitar se..."
7,87,G-224961 é seu código de verificação do Google.,1,0,"Entrega ou informa um código, sem solicitar se..."
8,88,Código de verificação do Google: G-647825. Não...,1,0,"Entrega ou informa um código, sem solicitar se..."
9,89,G-572293 é seu código de verificação do Google.,1,0,"Entrega ou informa um código, sem solicitar se..."


## Pendências e limites
As demais divergências não foram todas adjudicadas individualmente. Não preencher decisões humanas a partir desta triagem automaticamente.

In [4]:
triagem=pd.read_csv(PASTA/'triagem_divergencias.csv')
display(triagem.situacao_analise.value_counts())
assert len(triagem)==resumo['antes']
assert len(auth)==121
assert (auth.valor_sugerido==auth.valor_gerado_revisado.astype(float)).all()
print('Contagens reconciliadas; fonte original preservada na execução do classificador.')

Contagens reconciliadas; fonte original preservada na execução do classificador.


situacao_analise
revisao_manual_pendente          2388
diferenca_metodologica            627
resultado_recalculado             525
validacao_ferramenta_pendente     391
conferir_criterio_estrutural      173
provavel_erro_original            145
falha_script_corrigida             11
Name: count, dtype: int64

Validação: as quatro células foram executadas em ordem via IPython no mesmo processo. O ambiente bloqueou a abertura de portas do kernel Jupyter; não houve execução por um kernel separado.